### ISSUE TITLE
double free or corruption (out)

### ISSUE THREAD

**Issue by Guoc0529** [Unsigned-Long/River#3]
“Bro, I successfully compiled RIVER and RIS-CALIB in Docker, but encountered the same issue for both: 'double free or corruption (out)'. Hope you can provide some guidance!”
![2024-04-12 14-25-27屏幕截图](https://github.com/Unsigned-Long/River/assets/143497038/eea6a42f-1837-4914-bf8f-3c1d91f9ebce)

**Comment by Unsigned-Long** [comment]
Can you point out the specific location of this error? I saw a warning above the error message in the picture that shouldn't appear: "_Framebuffer with requested attributes not available, Using available framebuffer..._". This warning does not come from our library.

I searched and found that it comes from the visualization library [Pangolin](https://github.com/stevenlovegrove/Pangolin.git). And this problem is also explained in its readme. Hope this helps!

![image](https://github.com/Unsigned-Long/River/assets/76953144/b0e4cd29-94ec-44c8-be75-a5423ad7aab6)

**Comment by Guoc0529** [comment]
I've identified that the error occurred here
![Uploading 2024-04-12 17-09-06屏幕截图.png…]()

**Comment by Guoc0529** [comment]
I've identified that the error occurred here
![Uploading 2024-04-12 17-09-06屏幕截图.png…]()

**Comment by Guoc0529** [comment]
![Uploading 2024-04-12 17-09-06屏幕截图.png…]()

**Comment by Guoc0529** [comment]
![image](https://github.com/Unsigned-Long/River/assets/143497038/ab6d650e-9a37-415d-b2bb-4a5031b2983a)
I've identified that the error occurred here

**Comment by Unsigned-Long** [comment]
This doesn't seem to make sense. We don't have any problems with our computer at all. What do you think is the problem? :sob:

**Comment by Unsigned-Long** [comment]
Moreover, you said there is a problem with the code call here, but in fact, this code only exists in river and not in ris-calib. But the strange thing is that you have this problem in both libraries, so I don't think there is a problem here. Probably somewhere else.

**Comment by YXT-yang** [comment]
"Hi bro, I had a similar problem once. I solved the problem by modifying the CMakeLists.txt file. Modify lines 10 and 11 as follows:"
Before modification
``` cmake
set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -march=native")
set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=native")
```
After modification
``` cmake
set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
```

**Comment by Guoc0529** [comment]
> “嗨兄弟，我曾经遇到过类似的问题。我通过修改CMakeLists.txt文件解决了该问题。修改第10行和第11行如下：” 修改前
> 
> ```cmake
> set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -march=native")
> set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=native")
> ```
> 
> 修改后
> 
> ```cmake
> set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
> set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
> ```

Bro, you're really talented!"

**Comment by Unsigned-Long** [comment]
Awesome, problem solved :smiley: . 

The `-march=native` option in **GCC compiler** is used to automatically set compilation flags based on the features supported by your **CPU**. If you're using an **Intel** or **AMD** CPU and a **GCC version 4.2.3** or higher, it's recommended to use `-march=native`.

This option adjusts the compilation settings according to your CPU architecture and instruction set extensions, optimizing the generated machine code. It selects the best instruction set based on your CPU's capabilities, which can improve program performance.

In summary, `-march=native` is a convenient option that automatically optimizes compilation results based on your hardware environment  .
